bazel/scripts/run_integ_tests.sh

```diff
@@ -0,0 +1,239 @@
+    echo "Building..."
+    bazel build "//lte/gateway/python/integ_tests/s1aptests:test_restore_mme_config_after_sanity" --define=on_magma_test=1
+    echo "Executing..."
+    sudo "${MAGMA_ROOT}/bazel-bin/lte/gateway/python/integ_tests/s1aptests/test_restore_mme_config_after_sanity"
+    echo "Cleanup finished successfully."
+}
+
+run_test() {
+    local TARGET=$1
+    local TARGET_PATH=${TARGET%:*}
+    local SHORT_TARGET=${TARGET#*:}
+    (
+        set -x
+        echo "BUILDING TEST: ${TARGET}"
+        bazel build "${TARGET}" --define=on_magma_test=1
+        echo "RUNNING TEST: ${TARGET}"
+        sudo "${MAGMA_ROOT}/bazel-bin/${TARGET_PATH}/${SHORT_TARGET}"
+    )
+}
+
+print_summary() {
+    local NUM_SUCCESS=$1
+    local TOTAL_TESTS=$2
+    echo "SUMMARY: ${NUM_SUCCESS}/${TOTAL_TESTS} tests were successful."
+    for TARGET in "${!TEST_RESULTS[@]}"
+    do
+        echo "  ${TARGET}: ${TEST_RESULTS[${TARGET}]}"
+    done
+}
+
+###############################################################################
+# SCRIPT SECTION
+###############################################################################
+
+TARGET_PATH="${1:-}"
+
+if [[ "${TARGET_PATH}" == "--help" ]];
+then
+    help
+    exit 0
+fi
+
+declare -a EXTENDED_TEST_TARGETS
+declare -A TEST_RESULTS
+NUM_SUCCESS=0
+NUM_RUN=1
+
+cd "${MAGMA_ROOT}"
+
+EXTENDED_TEST_SETUP="//lte/gateway/python/integ_tests/s1aptests:test_modify_mme_config_for_sanity"
+EXTENDED_TEST_TEARDOWN="//lte/gateway/python/integ_tests/s1aptests:test_restore_mme_config_after_sanity"
+SKIP_EXTENDED_SETUP_AND_TEARDOWN="false"
+
+if [[ "${TARGET_PATH}" == "--list" ]];
+then
+    list_all_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--list-extended" ]];
+then
+    list_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--list-traffic-server" ]];
+then
+    list_traffic_server_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--setup-extended" || "${TARGET_PATH}" == "${EXTENDED_TEST_SETUP}" ]];
+then
+    setup_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--teardown-extended" || "${TARGET_PATH}" == "${EXTENDED_TEST_TEARDOWN}" ]];
+then
+    teardown_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--skip-setup-teardown-extended" ]];
+then
+    TARGET_PATH="${2:-}"
+    SKIP_EXTENDED_SETUP_AND_TEARDOWN="true"
+fi
+
+create_test_targets
+
+TOTAL_TESTS=${#EXTENDED_TEST_TARGETS[@]}
+# TODO TOTAL_TESTS=$((TOTAL_TESTS + PRECOMMIT))
+
+if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
+then
+    setup_extended_tests
+fi
+
+for TARGET in "${EXTENDED_TEST_TARGETS[@]}"
+do
+    echo "Starting test ${NUM_RUN}/${TOTAL_TESTS}: ${TARGET}"
+    if run_test "${TARGET}";
+    then
+        NUM_SUCCESS=$((NUM_SUCCESS + 1))
+        TEST_RESULTS["${TARGET}"]="PASSED"
+    else
+        TEST_RESULTS["${TARGET}"]="FAILED"
+    fi
+    NUM_RUN=$((NUM_RUN + 1))
+done
+
+if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
+then
+    teardown_extended_tests
```

Comment:
When running all tests the teardown seems to fail and I don't get to see the test summary.
![Screenshot from 2022-07-22 15-51-15](https://user-images.githubusercontent.com/34488763/180453581-2dd00fa6-b22f-412b-8e96-1050c2f188b5.png)

- Maybe it would be better to first print the summary and then attempt the teardown?
- Do we know why the teardown fails in this case?

Reply:
I would guess that you deleted the backup file or stashed it in git during the long execution. The teardown will only run if there was a setup before, except when you execute the teardown direcly, but then you do not have a summary to worry about. I added an extra check, to only clean up if there is a backup file. I do not think putting it after the summary is worth it, this way you do not have to scroll for the results.